Adds Eucharist and Accompanying Features for Chaplains

XanderDox opened the pull request:
**Adds Eucharist Wafers**

- Adds a Eucharist wafer item. This cannot be prepared in the kitchen, instead Chaplains with the Christianity religion can use a bible to consecrate crackers (which can be prepared by the kitchen already) into wafers, similar to how they can bless water.

- ~Chaplains with the Christianity religion can click on the wafer to raise it into the air for nearby people to see (primarily for religious services). This is separate from the 'show held object' to help differentiate this ritualistic action from simply holding up a crowbar~ (removed at dev request)

- Anyone eating the wafer will taste 'holy bread'.

<img width="115" height="133" alt="image" src="https://github.com/user-attachments/assets/ae28f2f1-9409-48be-ad7f-7caf900ebf04" />

**Adds Pyx**

- Adds a 'Pyx', a small item used by clergy to transport the Eucharist to people unable to attend Mass. It can hold three wafers, and has open, closed, and filled sprites.

<img width="341" height="133" alt="image" src="https://github.com/user-attachments/assets/54c12a55-39f7-41f5-8382-aa5ae64d1ce3" />

:cl: XanderDox
rscadd: Adds Eucharist wafers, which can be consecrated from crackers by Christian Chaplains.
rscadd: Adds pyx for transporting Eucharist wafers to people.
imageadd: Adds a sprite for pyx, including open, closed, and filled, courtesy of Keekenox/Slimex.
imageadd: Adds a sprite for Eucharist wafers.
/:cl:

CakeQ commented on code/modules/reagents/reagent_containers/food/snacks.dm:
```diff
@@ -2361,6 +2361,41 @@
+				qdel(src)
+				return TRUE
+			else
+				to_chat(user, "Nothing happened.")
```
This is unnecessary, just leave it to the ..() to handle

CakeQ commented on code/modules/reagents/reagent_containers/food/snacks.dm:
```diff
@@ -2361,6 +2361,41 @@
 	nutriment_amt = 1
 
+/obj/item/reagent_containers/food/snacks/cracker/use_tool(obj/item/W, mob/living/carbon/human/user, list/click_params)
+	. = FALSE
```
This will never get used, remove it

CakeQ commented on code/modules/reagents/reagent_containers/food/snacks.dm:
```diff
@@ -2361,6 +2361,41 @@
+/obj/item/reagent_containers/food/snacks/eucharist/attack_self(mob/living/carbon/human/user)
+	if(!ishuman(user))
+		return
+	if(user.mind && istype(user.mind.assigned_job, /datum/job/chaplain))
```
Similar to your other PR, I'm not really sure this code block is needed with pre-existing procs such as show-held-item, but if you're insistent on keeping this in the PR, remove the job check. You don't need to be assigned a chaplain by the social-economic structure in order to be able to raise a wafer in the air.

XanderDox replied:
Raising the wafer in a specific manner is a core-part of Christian ritual, and this follows the pre-existing standard of the bible's interactions being limited behind the Chaplain role. It protects their mechanics which otherwise are not protected by the Skills System or rules, and makes the role feel more unique than a passenger.

That being said, I am removing it as requested.

HeyBanditoz commented on code/modules/reagents/reagent_containers/food/snacks.dm:
```diff
@@ -2361,6 +2361,31 @@
+	if(user.mind && istype(user.mind.assigned_job, /datum/job/chaplain))
+		if(istype(W,/obj/item/storage/bible/bible))
+			var/singleton/cultural_info/religion = user.get_cultural_value(TAG_RELIGION)
+			if(religion.name == RELIGION_CHRISTIANITY)
```
nit: style
```suggestion
	if (user.mind && istype(user.mind.assigned_job, /datum/job/chaplain))
		if (istype(W,/obj/item/storage/bible/bible))
			var/singleton/cultural_info/religion = user.get_cultural_value(TAG_RELIGION)
			if (religion.name == RELIGION_CHRISTIANITY)
```

XanderDox replied:
Fixed!

XanderDox committed:
Adds Eucharist Wafers and Pyx.

Adds Eucharist wafers that can be consecrated by Chaplains with a bible from crackers. Adds a Pyx that can carry around three wafers, it is in the loadout for chaplains. Pyx sprite courtesy of Keekenox/Slimex

HeyBanditoz commented on code/modules/reagents/reagent_containers/food/snacks.dm:
```diff
@@ -2361,6 +2361,31 @@
+				to_chat(user, SPAN_NOTICE("You consecrate the wafer."))
+				qdel(src)
+				return TRUE
+			else
```
Empty `else` block can be nuked

XanderDox replied:
Fixed

HeyBanditoz commented on code/game/objects/items/weapons/storage/pyx.dm:
```diff
@@ -0,0 +1,22 @@
+	icon_state = "[initial(icon_state)][opened ? "0" : ""]"
+	if (!opened)
+		return
+	for (var/i = 1 to length(contents))
+		if (istype(contents[i], /obj/item/reagent_containers/food/snacks/eucharist))
+			AddOverlays(image(icon, "pyx1"))
```
more succinctly:
also, adding `break` will prevent multiple overlays from being added, in case that can happen
```suggestion
	for (var/obj/item/reagent_containers/food/snacks/eucharist/item in contents)
	    AddOverlays(image(icon, "pyx1"))
	    break
```

XanderDox replied:
Fixed

HeyBanditoz commented on code/game/objects/items/weapons/storage/pyx.dm:
```diff
@@ -0,0 +1,22 @@
+/obj/item/storage/fancy/pyx/
```
```suggestion
/obj/item/storage/fancy/pyx
```

XanderDox replied:
Fixed

HeyBanditoz reviewed (dismissed):
this is the LAST one!!! I SWEAR!
